lib/IDE/CodeCompletion.cpp

```diff
@@ -2667,17 +2667,19 @@ class CompletionLookup final : public swift::VisibleDeclConsumer {
       Builder.addRightParen();
     }
 
-    // Enum element is of function type such as EnumName.type -> Int ->
-    // EnumName; however we should show Int -> EnumName as the type
-    Type EnumType;
-    if (EED->hasInterfaceType()) {
-      EnumType = EED->getInterfaceType();
-      if (auto FuncType = EnumType->getAs<AnyFunctionType>()) {
+    if (!EED->hasInterfaceType())
+      return;
+
+    // Enum element is of function type; (Self.type) -> Self or
+    // (Self.Type) -> (Args...) -> Self. We should show only 'Self' part.
+    Type EnumType = EED->getInterfaceType();
+    if (auto FuncType = EnumType->getAs<AnyFunctionType>()) {
+      EnumType = FuncType->getResult();
+      if (auto FuncType = EnumType->getAs<FunctionType>())
         EnumType = FuncType->getResult();
```

Comment:
The result type of an enum constructor is always the type of the enum itself. Why not just use that instead of looking at EED->getInterfaceType()?

Reply:
ie, just call `ED->getDeclaredInterfaceType()` where ED is your EnumDecl

Reply:
Not strong reason. But we want to display bound generic types for parameters as well as the result enum type. To do that, I want to use `getTypeOfMember()` because it can substitute generic parameters all at once.

I added another commit for another commit for displaying bound generic types. Could you take a look?